### PR TITLE
[tests-only] added unit tests for accordion item

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- added import for components used inside
    > testing without import fills console with vue warn errors, see details ⬇️ 
  <details>
    ```console

        [Vue warn]: Unknown custom element: <oc-button> - did you register the component correctly? For recursive components, make sure to provide the "name" option.

        found in

        ---> <OcAccordionItem>
           <Root>

          at warn (node_modules/vue/dist/vue.runtime.common.dev.js:621:15)
          at createElm (node_modules/vue/dist/vue.runtime.common.dev.js:5918:11)
          at createChildren (node_modules/vue/dist/vue.runtime.common.dev.js:6033:9)
          at createElm (node_modules/vue/dist/vue.runtime.common.dev.js:5934:9)
          at createChildren (node_modules/vue/dist/vue.runtime.common.dev.js:6033:9)
          at createElm (node_modules/vue/dist/vue.runtime.common.dev.js:5934:9)
          at VueComponent.patch [as __patch__] 
        (node_modules/vue/dist/vue.runtime.common.dev.js:6455:7)
</details>

- added tests for OcAccordion Component


## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Part of #1366 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
<!-- Tick the checkboxes when done. -->
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:
<!-- In case of incomplete PR, please list the open tasks here -->
- [ ] ...

### CODE REVIEW

**Review by individual-it — COMMENTED** [review]
@kulmann what do you think about the tests? testing too much implementation details?

File: src/components/OcAccordionItem.spec.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    describe('when icon is given', ()=> {
+      it('should render an icon on the left side of title', () => {
+        const wrapper = getWrapperWithProps({
+          icon: 'mdi-icon'
+        })
+        const titleHeadingSpan = wrapper.get('span.uk-width-1-1')
+        expect(titleHeadingSpan.findAll('.oc-mr-s').length).toBe(1)
+        expect(titleHeadingSpan).toMatchSnapshot()
+      })
+      it ('should render extra space before description if given', () => {
```
Comment (kulmann) [inline src/components/OcAccordionItem.spec.js:68]: Too much level of detail. We could later on introduce visual regression tests for things like spacing, but please omit such things in unit tests. Unit tests should be about functionality, not so much about appearance.

**Review by kulmann — CHANGES_REQUESTED** [review]
The tests themselves achieve want you wanted to test - but the scope of the tests is different from what I would expect. I'd expect unit tests to test functionality. E.g. if all the props and their input value variations lead to the expected behaviour of the component and if interaction with the component leads to the expected events to be emitted. Please don't test visual attributes (like spacings, icon sizes, etc). We don't have visual regression tests here, yet, and I don't want to introduce them right now, but later on we could introduce VRT to check the visual appearance of components - that would include proper spacing, icon sizes, etc, then.

File: src/components/OcAccordionItem.spec.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        })
+        const titleHeadingSpan = wrapper.get('span.uk-width-1-1')
+        expect(titleHeadingSpan.findAll('.oc-mr-s').length).toBe(2)
+        expect(titleHeadingSpan).toMatchSnapshot()
+      })
+    })
+  })
+  describe('toggle expanded', () => {
+    // https://forum.vuejs.org/t/test-this-parent-emit/42688/2
+    // cannot test if parent emits or not form the child component
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/components/OcAccordionItem.spec.js]: fixed

File: src/components/OcAccordionItem.spec.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    // only checks if the toggle method is called
+    it('when title button is clicked', async () => {
+      const wrapper = getWrapperWithMockMethods()
+      const titleButton = wrapper.find('.uk-text-left.uk-width-1-1')
+      await titleButton.trigger('click')
+      await wrapper.vm.$nextTick()
+      expect(mockMethods.toggleExpanded).toHaveBeenCalledTimes(1)
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/components/OcAccordionItem.spec.js]: I've updated the emitter to emit from `OcAccordionItem` component.
Does the `mounted()` func in `OcAccordionItem` catches those emitted event?

File: src/components/OcAccordionItem.spec.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      methods: mockMethods
+    })
+  }
+  describe('default behaviours', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({})
+    })
+    it('should set item id as default', () => {
+      expect(wrapper.attributes('id')).toContain('oc-accordion-id-')
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/components/OcAccordionItem.spec.js]: fixed

File: src/components/OcAccordionItem.spec.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    })
+    it('should render title as heading with default heading level', () => {
+      const titleElement = wrapper.find(`h${defaultHeadingLevel}`)
+      expect(titleElement.exists()).toBeTruthy()
+      expect(titleElement.attributes('class')).toBe('oc-accordion-title')
+      expect(titleElement.find('.uk-width-expand').text()).toBe('Test title')
+    })
+  })
+  it('should render description if given', () => {
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/components/OcAccordionItem.spec.js]: fixed

File: src/components/OcAccordionItem.spec.js (hunk 1 above)
Comment (kiranparajuli589) [reply src/components/OcAccordionItem.spec.js]: fixed

File: src/components/OcAccordionItem.spec.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  describe('default behaviours', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({})
+    })
+    it('should set item id as default', () => {
+      expect(wrapper.attributes('id')).toContain('oc-accordion-id-')
+    })
+    it('should render title as heading with default heading level', () => {
+      const titleElement = wrapper.find(`h${defaultHeadingLevel}`)
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/components/OcAccordionItem.spec.js]: fixed

File: src/components/OcAccordionItem.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/components/OcAccordionItem.spec.js:90]: I tried using inline bindings for slots but unfortunately, that is also not possible
https://github.com/vuejs/vue/issues/4332#issuecomment-263444492
IMO, this is a developer thing to look at.
I've reverted the change made to the emitter part in OcAccortionItem component

**Review by individual-it — COMMENTED** [review]
@kulmann what do you want to do with this?
I would suggest to merge the tests and open an issue with the needed refactoring

**Review by kulmann — APPROVED** [review]

**Pull request merged**